> Set `<IsNETCoreAppRef>false</IsNETCoreAppRef>` [here](https://github.com/dotnet/runtime/blob/fa86c81b61b10daf3da4438b894a7d4405e229e8/src/libraries/System.Net.Quic/ref/System.Net.Quic.csproj#L5)

I just tried that and I got compilations errors:
```
System.Text.Json -> /home/manicka/repositories/runtime/artifacts/bin/System.Text.Json/net6.0-Debug/System.Text.Json.dll
/usr/share/dotnet/sdk/6.0.100-preview.4.21255.9/Microsoft.Common.CurrentVersion.targets(2202,5): error MSB3245: Could not resolve this reference. Could not locate the assembly "System.Net.Quic". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/home/manicka/repositories/runtime/src/libraries/System.Net.Http/src/System.Net.Http.csproj]        
  Microsoft.VisualBasic.Core -> /home/manicka/repositories/runtime/artifacts/bin/Microsoft.VisualBasic.Core/net6.0-Debug/Microsoft.VisualBasic.Core.dll
/home/manicka/repositories/runtime/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs(6,18): error CS0234: The type or namespace name 'Quic' does not exist in the namespace 'System.Net' (are you missing an assembly reference?) [/home/manicka/repositories/runtime/src/libraries/System.Net.Http/src/System.Net.Http.csproj]     
/home/manicka/repositories/runtime/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs(7,18): error CS0234: The type or namespace name 'Quic' does not exist in the namespace 'System.Net' (are you missing an assembly reference?) [/home/manicka/repositories/runtime/src/libraries/System.Net.Http/src/System.Net.Http.csproj] 
...
```

I assume that the ref is now not produced, but that means that even our own libraries cannot see it?

--- a/src/libraries/System.Net.Http/ref/System.Net.Http.cs
+++ b/src/libraries/System.Net.Http/ref/System.Net.Http.cs
@@ -388,7 +388,6 @@ namespace System.Net.Http
         public bool EnableMultipleHttp2Connections { get { throw null; } set { } }
         public Func<SocketsHttpConnectionContext, System.Threading.CancellationToken, System.Threading.Tasks.ValueTask<System.IO.Stream>>? ConnectCallback { get { throw null; } set { } }
         public Func<SocketsHttpPlaintextStreamFilterContext, System.Threading.CancellationToken, System.Threading.Tasks.ValueTask<System.IO.Stream>>? PlaintextStreamFilter { get { throw null; } set { } }
-        public System.Net.Quic.Implementations.QuicImplementationProvider? QuicImplementationProvider { get { throw null; } set { } }
     }
     public sealed class SocketsHttpConnectionContext
     {

--- a/src/libraries/System.Net.Http/src/System.Net.Http.csproj
+++ b/src/libraries/System.Net.Http/src/System.Net.Http.csproj
@@ -629,7 +629,7 @@
     <Reference Include="System.Net.NameResolution" />
     <Reference Include="System.Net.NetworkInformation" />
     <Reference Include="System.Net.Primitives" />
-    <Reference Include="System.Net.Quic" />
+    <ProjectReference Include="..\..\System.Net.Quic\ref\System.Net.Quic.csproj" />
     <Reference Include="System.Net.Security" />
     <Reference Include="System.Net.Sockets" />
     <Reference Include="System.Runtime" />

--- a/src/libraries/System.Net.Http/src/System.Net.Http.csproj
+++ b/src/libraries/System.Net.Http/src/System.Net.Http.csproj
@@ -616,7 +616,7 @@
              Link="Common\System\Net\Http\HttpHandlerDefaults.cs" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Net.Quic\ref\System.Net.Quic.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Net.Quic\src\System.Net.Quic.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Principal.Windows\src\System.Security.Principal.Windows.csproj" />
     <Reference Include="Microsoft.Win32.Primitives" />
     <Reference Include="System.Collections" />

--- a/src/libraries/System.Net.Http/src/System.Net.Http.csproj
+++ b/src/libraries/System.Net.Http/src/System.Net.Http.csproj
@@ -616,6 +616,7 @@
              Link="Common\System\Net\Http\HttpHandlerDefaults.cs" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Net.Quic\ref\System.Net.Quic.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Principal.Windows\src\System.Security.Principal.Windows.csproj" />
     <Reference Include="Microsoft.Win32.Primitives" />
     <Reference Include="System.Collections" />
@@ -629,7 +630,6 @@
     <Reference Include="System.Net.NameResolution" />
     <Reference Include="System.Net.NetworkInformation" />
     <Reference Include="System.Net.Primitives" />
-    <ProjectReference Include="..\..\System.Net.Quic\ref\System.Net.Quic.csproj" />
     <Reference Include="System.Net.Security" />
     <Reference Include="System.Net.Sockets" />
     <Reference Include="System.Runtime" />

--- a/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/SocketsHttpHandler.cs
+++ b/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/SocketsHttpHandler.cs
@@ -188,11 +188,5 @@ namespace System.Net.Http
             get => throw new PlatformNotSupportedException();
             set => throw new PlatformNotSupportedException();
         }
-
-        public QuicImplementationProvider? QuicImplementationProvider
-        {
-            get => throw new PlatformNotSupportedException();
-            set => throw new PlatformNotSupportedException();
-        }
     }
 }

--- a/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs
+++ b/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs
@@ -394,21 +394,6 @@ namespace System.Net.Http
             }
         }
 
-        /// <summary>
-        /// Gets or sets the QUIC implementation to be used for HTTP3 requests.
-        /// </summary>
-        public QuicImplementationProvider? QuicImplementationProvider
-        {
-            // !!! NOTE !!!
-            // This is temporary and will not ship.
-            get => _settings._quicImplementationProvider;
-            set
-            {
-                CheckDisposedOrStarted();
-                _settings._quicImplementationProvider = value;
-            }
-        }
-
         public IDictionary<string, object?> Properties =>
             _settings._properties ?? (_settings._properties = new Dictionary<string, object?>());
 

--- a/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTestBase.SocketsHttpHandler.cs
+++ b/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTestBase.SocketsHttpHandler.cs
@@ -49,7 +49,7 @@ namespace System.Net.Http.Functional.Tests
             if (quicImplementationProvider != null)
             {
                 SocketsHttpHandler socketsHttpHandler = (SocketsHttpHandler)GetUnderlyingSocketsHttpHandler(handler);
-                socketsHttpHandler.QuicImplementationProvider = quicImplementationProvider;
+                SetQuicImplementationProvider(socketsHttpHandler, quicImplementationProvider);
                 socketsHttpHandler.SslOptions.RemoteCertificateValidationCallback = (sender, certificate, chain, errors) => true;
             }
 
@@ -66,11 +66,18 @@ namespace System.Net.Http.Functional.Tests
         protected static HttpClientHandler CreateHttpClientHandler(string useVersionString) =>
             CreateHttpClientHandler(Version.Parse(useVersionString));
 
-
         protected static SocketsHttpHandler GetUnderlyingSocketsHttpHandler(HttpClientHandler handler)
         {
             FieldInfo field = typeof(HttpClientHandler).GetField("_underlyingHandler", BindingFlags.Instance | BindingFlags.NonPublic);
             return (SocketsHttpHandler)field?.GetValue(handler);
+        }
+
+        protected static void SetQuicImplementationProvider(SocketsHttpHandler handler, QuicImplementationProvider quicImplementationProvider)
+        {
+            FieldInfo settingsField = typeof(SocketsHttpHandler).GetField("_settings", BindingFlags.Instance | BindingFlags.NonPublic);
+            object settings = settingsField.GetValue(handler);
+            FieldInfo field = settings.GetType().GetField("_quicImplementationProvider", BindingFlags.Instance | BindingFlags.NonPublic);
+            field.SetValue(settings, quicImplementationProvider);
         }
 
         protected static HttpRequestMessage CreateRequest(HttpMethod method, Uri uri, Version version, bool exactVersion = false) =>

--- a/src/libraries/System.Net.Http/tests/FunctionalTests/System.Net.Http.Functional.Tests.csproj
+++ b/src/libraries/System.Net.Http/tests/FunctionalTests/System.Net.Http.Functional.Tests.csproj
@@ -254,6 +254,7 @@
   <ItemGroup>
     <PackageReference Include="System.Net.TestData" Version="$(SystemNetTestDataVersion)" />
     <ProjectReference Include="$(LibrariesProjectRoot)System.DirectoryServices.Protocols\src\System.DirectoryServices.Protocols.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Net.Quic\src\System.Net.Quic.csproj" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="$(CommonTestPath)StreamConformanceTests\StreamConformanceTests.csproj" />

--- a/src/libraries/System.Net.Quic/ref/System.Net.Quic.csproj
+++ b/src/libraries/System.Net.Quic/ref/System.Net.Quic.csproj
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
     <Nullable>enable</Nullable>
+    <IsNETCoreAppRef>false</IsNETCoreAppRef>
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="System.Net.Quic.cs" />

